src/library/scala/collection/IterableOnce.scala

```diff
@@ -1055,7 +1055,7 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
   def toVector: immutable.Vector[A] = immutable.Vector.from(this)
 
   def toMap[K, V](implicit ev: A <:< (K, V)): immutable.Map[K, V] =
-    immutable.Map.from(this.asInstanceOf[IterableOnce[(K, V)]])
+    immutable.Map.from(ev.substituteCo(this))
```

Comment:
This adds a few indirections. Compared to the cost of creating a map it shouldn't matter though.

Reply:
We have these casts in several spots and they are a conscious choice to avoid actually calling the conversion operation (which is guaranteed to be an identity cast anyway). I'd prefer to keep it that way and eventually remove the `ev` value entirely in Dotty, using `erased` instead of `implicit`.

Reply:
Maybe the conscious casts should be annotated with comments stating the above, or better yet a link to GitHub issue to (eventually) remove `ev`?

Reply:
Or annotate with annotation. There's a PR not to warn when `implicit <:<` is unused, but possible migration would be `@erased("in future, obviate the cast seen here")`.